src/main/java/io/anserini/util/TrainingDataGenerator.java

```diff
@@ -0,0 +1,360 @@
+   * This logger is used to write training data.
+   *
+   * @param loggerName     the name of the logger to create
+   * @param outputFilePath the file path to write logs to
+   * @param patternLayout  layout for the logger, if null just display the message using DEFAULT_CONVERSION_PATTERN
+   */
+  private static void createNewLoggerConfig(String loggerName,
+                                            String outputFilePath,
+                                            String patternLayout) {
+    // Create a logger to write the training data
+    final LoggerContext ctx = (LoggerContext) LogManager.getContext(false);
+    final Configuration config = ctx.getConfiguration();
+
+    Layout layout = PatternLayout.createLayout(patternLayout == null ? PatternLayout.DEFAULT_CONVERSION_PATTERN : patternLayout,
+            config, null, StandardCharsets.UTF_8, false,
+            false, null, null);
+
+    Appender appender = FileAppender.createAppender(outputFilePath,
+            "false", "false", loggerName, "true",
+            "false", "false", "2000",
+            layout, null, "false", null, config);
+    appender.start();
+
+    config.addAppender(appender);
+
+    // Adding reference to the appender
+    AppenderRef ref = AppenderRef.createAppenderRef(loggerName, null, null);
+    AppenderRef[] refs = new AppenderRef[]{ref};
+
+    LoggerConfig loggerConfig = LoggerConfig.createLogger("false", Level.TRACE,
+            loggerName, "true", refs, null, config, null);
+
+    // Adding appender to logger, and adding logger to context
+    loggerConfig.addAppender(appender, null, null);
+
+    config.addLogger(loggerName, loggerConfig);
+    ctx.updateLoggers();
+  }
+
+  private void initializeIndex() throws IOException {
+    if (kbIndexDirectory == null && args.index != null) {
+      LOG.info("Initializing KB index from directory: {}", args.index);
+      kbIndexDirectory = FSDirectory.open(Paths.get(args.index));
+    }
+  }
+
+  /**
+   * Gets index reader from the directory specified in the user parameters
+   *
+   * @return KB index reader
+   * @throws IOException on error
+   */
+  private IndexReader getKbIndexReader() throws IOException {
+    if (kbIndexReader == null) {
+      initializeIndex();
+      kbIndexReader = DirectoryReader.open(kbIndexDirectory);
+    }
+
+    return kbIndexReader;
+  }
+
+  /**
+   * @return index searcher
+   * @throws IOException on error
+   */
+  IndexSearcher getKbIndexSearcher() throws IOException {
+    if (kbIndexSearcher == null) {
+      kbIndexSearcher = new IndexSearcher(getKbIndexReader());
+    }
+
+    return kbIndexSearcher;
+  }
+
+  /**
+   * Get the current default index analyzer or create it.
+   * Use EnglishAnalyzer as the default.
+   *
+   * @return kb index analyzer
+   * @throws Exception
+   */
+  Analyzer getKbIndexAnalyzer() throws Exception {
+    if (kbIndexAnalyzer == null) {
+      kbIndexAnalyzer = new EnglishAnalyzer();
+    }
```

Comment:
Using `EnglishAnalyzer()` without any parameters assumes a default stopword list. If that's not okay, use `EnglishAnalyzer(CharArraySet.EMPTY_SET)` instead.